python/plugins/db_manager/dlg_create_table.py

```diff
@@ -288,7 +291,7 @@ def createTable(self):
             geomType = self.GEOM_TYPES[self.cboGeomType.currentIndex()]
             geomDim = self.spinGeomDim.value()
             try:
-                geomSrid = int(self.editGeomSrid.text())
+                geomSrid = self.widgetTargetSrid.crs().postgisSrid()
```

Comment:
How does postgisSrid work with a custom/user projection? I would like to see us avoid this method wherever possible -- it has no place in the wonderful world of proj >= 6 and will likely be removed in QGIS 4.0. Rather, we should lookup the correct code here from the spatial_ref_sys table by looking up the authority and identifier (and falling back to lookup by proj string for user projections).

Reply:
As I can see it never worked with custom projections, as it was a simple line edit and entered value was casted to integer id before using (and falling back to 0 in case of failure)

Reply:
Right, the previous approach put the responsibility on the user to enter a valid postgis srs id. But by flipping to the user-friendly widget, QGIS is taking on that responsibility. And currently we aren't handling it correctly...